### PR TITLE
Improve some notification and translation docs

### LINKED ISSUE #89804: `NOTIFICATION_TRANSLATION_CHANGED` gets sent at wrong times
### Tested versions

- Reproducible in 4.3.dev4, 4.3.dev5
- Not reproducible in 4.3.dev3

Likely due to #87530 (@YeldhamDev)

### System information

Windows 10

### Issue description

The notification `NOTIFICATION_TRANSLATION_CHANGED` now gets emit when ~~creating nodes~~ nodes enter the tree. 

It happens before the node is ready, which leads to crashes in one of the most common use case: you hook something like `update_ui()` to it. Generally the function to update UI will set a bunch of text on other nodes, but as the node just got created and its children aren't ready, it will lead to crashes. This also creates duplicate calls to the `update_ui`, because in most uses `_ready` would already take care of initial setup call.

### Steps to reproduce

- Create a scene with node that listens on `_notification` for `NOTIFICATION_TRANSLATION_CHANGED`. (print something or put a breakpoint)
- The notification will happen when running project, even when locale did not change.

### Minimal reproduction project (MRP)

[Translation changed bug MRP.zip](https://github.com/godotengine/godot/files/14731563/Translation.changed.bug.MRP.zip)

### PR DESCRIPTION
**Pull request by KoBeWi**
Closes #89804

### CODE REVIEW

File: doc/classes/Node.xml [hunk 1]
```diff
@@ -1170,7 +1170,15 @@
 			Implemented only on iOS.
 		</constant>
 		<constant name="NOTIFICATION_TRANSLATION_CHANGED" value="2010">
-			Notification received when translations may have changed. Can be triggered by the user changing the locale. Can be used to respond to language changes, for example to change the UI strings on the fly. Useful when working with the built-in translation support, like [method Object.tr].
+			Notification received when translations may have changed. Can be triggered by the user changing the locale or when the node enters scene tree. Can be used to respond to language changes, for example to change the UI strings on the fly. Useful when working with the built-in translation support, like [method Object.tr].
```
Comment (AThousandShips) [inline doc/classes/Node.xml:1173]:
```suggestion
			Notification received when translations may have changed. Can be triggered by the user changing the locale or when the node enters the scene tree. Can be used to respond to language changes, for example to change the UI strings on the fly. Useful when working with the built-in translation support, like [method Object.tr].
```

File: doc/classes/Control.xml [hunk 2]
```diff
@@ -1181,6 +1181,14 @@
 			- One of the node's theme property overrides is changed.
 			- The node enters the scene tree.
 			[b]Note:[/b] As an optimization, this notification won't be sent from changes that occur while this node is outside of the scene tree. Instead, all of the theme item updates can be applied at once when the node enters the scene tree.
+			[b]Note:[/b] This notification is received alongside [constant NOTIFICATION_ENTER_TREE], so if you are instantiating a scene, the child nodes will not be initialized yet. You can use it to setup theming for this node, child nodes created from script, or if you want to access child nodes added in the editor, make sure the node is ready using [method Node.is_node_ready].
```
Comment (AThousandShips) [inline doc/classes/Control.xml:1184]:
```suggestion
			[b]Note:[/b] This notification is received alongside [constant Node.NOTIFICATION_ENTER_TREE], so if you are instantiating a scene, the child nodes will not be initialized yet. You can use it to setup theming for this node, child nodes created from script, or if you want to access child nodes added in the editor, make sure the node is ready using [method Node.is_node_ready].
```

**Review by AThousandShips — APPROVED** [review]

**Comment by YeldhamDev** [comment]
Before this is merged, I would like to tackle a way that makes the current behavior people use to update texts still work, will push a PR soon.

**Comment by Kiisu-Master** [comment]
On the `NOTIFICATION_TRANSLATION_CHANGED` doc also mention it gets sent when setting auto translate mode. It seems like a important thing and I kind of forgot about that. When creating the issue, I thought it only gets sent when changing locale or smth (as I have never had a reason to set the auto translate mode at runtime).

**Comment by YeldhamDev** [comment]
> I thought it only gets sent when changing locale or smth

The current naming definitely doesn't help. We discussed in the dev chat that also the opportunity arrives, to replace `_CHANGED` with `_UPDATED`.

Also, I'm afraid that people will have to adapt to the new way of doing it. As I can't think of a way to support the old behavior without over-complicating things.

**Comment by KoBeWi** [comment]
> On the NOTIFICATION_TRANSLATION_CHANGED doc also mention it gets sent when setting auto translate mode.

Done.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**